### ISSUE TITLE
Support install not in .git root

### ISSUE THREAD

**Issue by jods4** [alirezanet/Husky.Net#103]
### Details

Currently, husky checks that it's being installed in the repository root, i.e. cwd contains a `.git` folder.

Some repository contain more than a .net solution, it's not uncommon to have other folders at root such as docs, infra, front-end, etc. In our setup, dotnet solution and `.config` local tools live in a `Source` folder, under root.

The issue with this is that:
1. Local dotnet tool can't run outside `Source` because they're local and do not exist outside of that folder.
2. Local tool Husky can't run inside `Source` because it requires `.git` in the same folder where it runs.

As 1. and 2. are mutually exclusive, there's no working configuration :(

> [!NOTE]
> Husky -- the NPM project that inspired Husky.Net -- has a "How to" for this topic in its documentation:
> https://typicode.github.io/husky/how-to.html#project-not-in-git-root-directory
>
> Their solution is simply to change working directory appropriately in install script and hook.
> The blocker is that whereas `cd .. & husky` works in NPM scripts, there's no way parameter to change the working directory in `dotnet run`, so `husky init` can't be run in a different cwd.

### Suggested solution
Add a `--cwd` parameter to install command, so that the working directory can be changed before it runs:
```
dotnet husky install --cwd ../..
```
The reverse `cd` command should be inferred automatically (using `Path.GetRelativePath`) and introduced in hook script so that they run in the correct folder (otherwise `dotnet husky` & other local .net tool tasks will not be found).

**Comment by jods4** [comment]
Thanks for the quick answer!

`--dir` doesn't help. I'm ok with having the folder at the root next to `.git`, or lower in `Source`. Both would be fine.

`husky install` does not even run because of this safety guard: https://github.com/alirezanet/Husky.Net/blob/master/src/Husky/Cli/InstallCommand.cs#L73-L79

**Comment by bdarlt** [comment]
I have a monorep with both an Angular app (/web) and a .Net solution (/api).

What mostly works is installing Husky.Net into the root folder (with the .git directory).  You can use "cwd" in some way to pick the part of the monorep you want. 

The prepare script of the Angular App (/web) installs/activates husky.  I haven't figured out a way to "activate" Husky on the .NET side.  Missing a monorep tool, I suppose.

```json
{
  "tasks": [
    {
      "name": "dotnet-format",
      "cwd": "api",
      "group": "pre-commit",
      "command": "dotnet",
      "args": ["format", "--include", "${staged} --severity error"],
      "include": ["**/*.cs", "**/*.vb"]
    },
    {
      "name": "eslint",
      "group": "pre-commit",
      "pathMode": "absolute",
      "cwd": ".",
      "command": "npx",
      "args": ["lint-staged", "-r", "--cwd", "web"],
      "include": ["**/*.ts", "**/*.js"]
    },
    {
      "name": "prettier",
      "group": "pre-commit",
      "pathMode": "absolute",
      "cwd": ".",
      "command": "npx",
      "args": ["prettier", "--write", "${staged}"],
      "include": [
        "**/*.ts",
        "**/*.vue",
        "**/*.jsx",
        "**/*.js",
        "**/*.json",
        "**/*.yml",
        "**/*.css",
        "**/*.scss"
      ]
    },
    {
      "name": "Welcome",
      "output": "always",
      "command": "bash",
      "args": ["-c", "echo Nice work! 🥂"],
      "windows": {
        "command": "cmd",
        "args": ["/c", "echo Nice work! 🥂"]
      }
    }
  ]
}
```

**Comment by jods4** [comment]
> What mostly works is installing Husky.Net into the root folder

Yes, this is the requirement that I would like to see lifted with this issue.

Depending on your repo structure, which may be outside your control, it's not always ok to add .net tools at the root of the repo.

**Comment by oleksii-yurchenko-bhge** [comment]
Having the same issue. We use monorepo, and have several .net solutions inside. So, whenever I try to add pre-commit hook it says that cannot find .git folder

**Comment by doggy8088** [comment]
I understand. You initialize .NET local tools at the root of the repository. That makes sense.

I hadn't considered that. As you mentioned, it's a monorepo project. I want the .NET-related tools, including the Husky tool, to be installed only under the `DemoApp` folder.

In Git 2.25, the Sparse Checkout feature is commonly used in many monorepo projects. This means I might never see any folders other than the `DemoApp` folder for .NET backend developers.

Do you believe it's a valid reason or scenario?

**Comment by doggy8088** [comment]
I understand your concern, and it makes sense. I have accepted your idea and will change my current setup. Thanks for the discussion!
